src/pages/index.js

```diff
@@ -0,0 +1,293 @@
+  };
+
+  // Remove any previously attached handler from deleteForm if it exists
+  deleteForm.removeEventListener("submit", deleteForm._deleteHandler);
+
+  // Store the new handler in a property and attach it to the delete form
+  deleteForm._deleteHandler = newDeleteHandler;
+  deleteForm.addEventListener("submit", newDeleteHandler, { once: true });
+
+  // Open the delete modal
+  openModal(deleteModal);
+}
+
+// Profile edit form submission
+editProfileForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  const saveButton = editProfileForm.querySelector(".modal__submit-btn");
+  showLoading(saveButton, true, "Saving...", "Save");
+
+  const name = editProfileForm.querySelector("#profile-name-input").value;
+  const about = editProfileForm.querySelector(
+    "#profile-description-input"
+  ).value;
+
+  api
+    .editUserInfo({ name, about })
+    .then((userData) => {
+      document.querySelector(".profile__name").textContent = userData.name;
+      document.querySelector(".profile__description").textContent =
+        userData.about;
+      closeModal(editModal);
+    })
+    .finally(() => showLoading(saveButton, false, "Saving...", "Save"));
+});
+
+// New event listener to pre-fill the form when opening the profile modal
+const editProfileButton = document.querySelector(".profile__edit-btn");
+editProfileButton.addEventListener("click", () => {
+  const editProfileForm = document.querySelector("#edit-profile-form");
+  const inputList = Array.from(
+    editProfileForm.querySelectorAll(settings.inputSelector)
+  );
+  // Clear validation errors and reset form state
+  resetValidation(editProfileForm, inputList, settings);
+  // Populate inputs with the current profile data
+  document.querySelector("#profile-name-input").value =
+    document.querySelector(".profile__name").textContent;
+  document.querySelector("#profile-description-input").value =
+    document.querySelector(".profile__description").textContent;
+
+  openModal(editModal); // Open the modal after setting the input values
+});
+
+// Avatar form submission
+avatarForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  const saveButton = avatarForm.querySelector(".modal__submit-btn");
+  showLoading(saveButton, true);
+  const avatarLink = avatarForm.querySelector("#profile-avatar-input").value;
+
+  api
+    .editAvatarInfo(avatarLink)
+    .then((userData) => {
+      document.querySelector(".profile__avatar").src = userData.avatar;
+      closeModal(avatarModal);
+    })
+    .finally(() => showLoading(saveButton, false));
+});
+
+// Add card form submission
+cardForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  const saveButton = cardForm.querySelector(".modal__submit-btn");
+  showLoading(saveButton, true, "Saving...", "Save");
+  const name = cardForm.querySelector("#card-caption-input").value;
+  const link = cardForm.querySelector("#add-card-link-input").value;
```

Comment:
* It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine